html/includes/api_functions.inc.php

```diff
@@ -347,7 +347,7 @@ function list_devices()
         $param[] = $_SESSION['user_id'];
     }
     $devices = array();
-    $dev_query = "SELECT $select FROM `devices` AS d $join WHERE $sql GROUP BY d.`hostname` ORDER BY $order";
+    $dev_query = "SELECT dev.*, lat, lng FROM (SELECT $select FROM `devices` AS d $join WHERE $sql GROUP BY d.`hostname` ORDER BY $order) dev LEFT JOIN locations ON locations.location=dev.location";
```

Comment:
Revert this line back to how it was and use this patch instead:

https://p.libren.ms/view/raw/d23ae0fb